elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/PrintableRequest.scala

```diff
@@ -0,0 +1,18 @@
+package com.sksamuel.elastic4s.http
+
+import com.sksamuel.exts.Logging
+
+/**
+  * @tparam T the type of the request object to be printed
+  */
+abstract class PrintableRequest[T] extends Logging {
+
+  def endpoint(request: T): String
```

Comment:
What do you think about having params and method defined seperately so people don't have to parse out params if they want to use them.